azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+						return fmt.Errorf("Unable to disable Custom Domain HTTPS for Frontend Endpoint %q (Resource Group %q): %+v", frontendEndpointName, resourceGroup, err)
+					}
+				}
+			}
+		}
+	}
+
+	return resourceArmFrontDoorRead(d, meta)
+}
+
+func resourceArmFrontDoorFrontendEndpointEnableHttpsProvisioning(enableCustomHttpsProvisioning bool, frontDoorName string, frontendEndpointName string, resourceGroup string, customHTTPSConfiguration frontdoor.CustomHTTPSConfiguration, meta interface{}) error {
+	client := meta.(*ArmClient).frontdoor.FrontDoorsFrontendClient
+	ctx := meta.(*ArmClient).StopContext
+
+	if enableCustomHttpsProvisioning {
+		future, err := client.EnableHTTPS(ctx, resourceGroup, frontDoorName, frontendEndpointName, customHTTPSConfiguration)
+
+		if err != nil {
+			return fmt.Errorf("Error enabling Custom Domain HTTPS for Frontend Endpoint: %+v", err)
+		}
+		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Error waiting to enable Custom Domain HTTPS for Frontend Endpoint: %+v", err)
+		}
+	} else {
+		future, err := client.DisableHTTPS(ctx, resourceGroup, frontDoorName, frontendEndpointName)
+
+		if err != nil {
+			return fmt.Errorf("Error disabling Custom Domain HTTPS for Frontend Endpoint: %+v", err)
+		}
+		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return fmt.Errorf("Error waiting to disable Custom Domain HTTPS for Frontend Endpoint: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmFrontDoorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).frontdoor.FrontDoorsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["frontdoors"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Front Door %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Front Door %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if properties := resp.Properties; properties != nil {
+		if err := d.Set("backend_pool", flattenArmFrontDoorBackendPools(properties.BackendPools)); err != nil {
+			return fmt.Errorf("Error setting `backend_pool`: %+v", err)
+		}
+		if err := d.Set("enforce_backend_pools_certificate_name_check", flattenArmFrontDoorBackendPoolsSettings(properties.BackendPoolsSettings)); err != nil {
+			return fmt.Errorf("Error setting `enforce_backend_pools_certificate_name_check`: %+v", err)
+		}
+		d.Set("cname", properties.Cname)
+		if properties.EnabledState == frontdoor.EnabledStateEnabled {
```

Comment:
I don't think we need this if statement?
```
			d.Set("load_balancer_enabled", properties.EnabledState == frontdoor.EnabledStateEnabled)
```

should handel the false case too?

Reply:
Yep, fixed.